src/scenario/terrain.cpp

```diff
@@ -111,12 +111,12 @@ void cTerrain::import_legacy(legacy::terrain_type_type& old){
 	// This is a weird expression, but necessary, because all archetypes except lava lack a special ability
 	is_archetype = archetypes.count(picture) && (old.special == 0) == (old.picture != 404);
 	blockage = (eTerObstruct) old.blockage;
-	if(picture < 260){
+	if(picture>=0 && picture < 260){
 		combat_arena = arenas[picture];
 		ground_type = ground[picture];
 		trim_type = (eTrimType) trims[picture];
 		trim_ter = trim_ters[picture];
-	}else if(picture >= 400 && picture < 1000){
+	}else if(picture >= 400 && picture < 140+274){
```

Comment:
Why such a strange number as `140+274`? What's wrong with `1000`?

Reply:
Because we have only 274 arenas:
`	static const short arenas[274] = {`

Reply:
Ahh, I see.

Hmm. Legacy BoE had only 14 animated terrain graphics, so it's not as though this would break anything. If something used a graphic number between 414 and 1000, then that would be an error on the scenario's part.

So I guess this is fine.